common/_static/helpfulness.js

```diff
@@ -0,0 +1,345 @@
+			show: function ()
+			{
+				if (!this.closed && !this.completed && this.shouldShow && !this.$container.is(":visible"))
+				{
+					var documentHeight = $(document).height();
+					var windowHeight = $(window).height();
+					var enoughScrollbars = (documentHeight - windowHeight) > 200;
+
+					if ($(window).scrollTop() > 100 || !enoughScrollbars)
+					{
+						this.$container.slideDown('fast');
+					}
+				}
+			},
+
+			submitYes: function()
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitYes",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			submitNo: function (value)
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form-no").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitNo",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId,
+							reason: value
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			getStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					return localStorage.getItem("helpfulness-" + contentId);
+				}
+
+				return null;
+			},
+
+			setStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					try
+					{
+						localStorage.setItem("helpfulness-" + contentId, 1);
+						return true;
+					}
+					catch (e)
+					{
+						if (e.name === 'QUOTA_EXCEEDED_ERR' || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+						{
+							return false;
+						}
+						else
+						{
+							return false;
+						}
+					}
+				}
+
+				return false;
+			},
+
+			showMessage: function (message)
+			{
+				this.$content.children(".messages").hide();
+				this.$content.children(message).show();
+			}
+		};
+	};
+})(jQuery);
+
+$(function ()
+{
+	$.Helpfulness({
+		maxCharacters: 160,
```

Comment:
For the 160 character limit, we are aligning to the limit being enforced on https://azure.microsoft.com. A longer limit will likely encourage people to attempt to use the comments for technical support which is no bueno. The character limit is very visible and counts down so the responder can see the limit at the outset and see the remaining characters.